delfin/drivers/pure/flasharray/pure_flasharray.py

```diff
@@ -363,6 +364,134 @@ def reset_connection(self, context, **kwargs):
     def get_access_url():
         return 'https://{ip}'
 
+    def collect_perf_metrics(self, context, storage_id, resource_metrics,
+                             start_time, end_time):
+        metrics = []
+        if resource_metrics.get(constants.ResourceType.STORAGE):
+            arrays_id, arrays_name = self.get_array()
+            storage_metrics = self.get_metrics(
```

Comment:
Method `get_metrics` is too large, it is recommended that split it according to function.

Reply:
Has been modified